src/libFLAC/cpu.c

```diff
@@ -245,9 +245,16 @@ ppc_cpu_info (FLAC__CPUInfo *info)
 #define PPC_FEATURE2_ARCH_2_07		0x80000000
 #endif
 
-	if (getauxval(AT_HWCAP2) & PPC_FEATURE2_ARCH_3_00) {
+	long hwcaps;
+#ifdef __linux__
+	hwcaps = getauxval(AT_HWCAP2);
+#elif defined (__FreeBSD__)
+	elf_aux_info(AT_HWCAP2, &hwcaps, sizeof(hwcaps));
+#endif
```

Comment:
What about adding
```
#else
        #error Unsupported platform! Please add support for reading ppc hwcaps.
```
so the code doesn't proceed with uninitialized data if someone happens to build on a third ppc platform?

Reply:
OK, that will it easier for users of other systems.

Reply:
Thanks!